### PR TITLE
maestral: 0.4.2 -> 0.6.1

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

Upstream is rapidly improving.

Daily driver.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/development/python-modules/pyro5/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{ buildPythonPackage
+, fetchPypi
+, lib
+, serpent
+, pythonOlder
+, pytest
```
Comment (jonringer) [inline pkgs/development/python-modules/pyro5/default.nix]:
```suggestion
, pytestCheckHook
```

File: pkgs/development/python-modules/pyro5/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+}:
+
+let
+  # ignore network related tests, which fail in sandbox
+  ignoredTests = [ "StartNSfunc" "Broadcast" "GetIP" ];
```
Comment (jonringer) [inline pkgs/development/python-modules/pyro5/default.nix]: and move into the expression please
```suggestion
  disabledTests = [ "StartNSfunc" "Broadcast" "GetIP" ];
```

File: pkgs/development/python-modules/pyro5/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  propagatedBuildInputs = [ serpent ];
+
+  disabled = pythonOlder "3.6";
+
+  checkInputs = [ pytest ];
+
+  checkPhase = ''
+    pytest -k '${lib.concatMapStringsSep " and " (e: "not ${e}") ignoredTests}'
+  '';
```
Comment (jonringer) [inline pkgs/development/python-modules/pyro5/default.nix]: remove

File: pkgs/development/python-modules/pyro5/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "02i3g35hjrvkiyjaskm2r5zrp3sjgxs8himksb0f6n093vg3lhrl";
+  };
+
+  propagatedBuildInputs = [ serpent ];
+
+  disabled = pythonOlder "3.6";
+
+  checkInputs = [ pytest ];
```
Comment (jonringer) [inline pkgs/development/python-modules/pyro5/default.nix]:
```suggestion
  checkInputs = [ pytestCheckHook ];
```

File: pkgs/development/python-modules/pyro5/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  version = "5.6";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "02i3g35hjrvkiyjaskm2r5zrp3sjgxs8himksb0f6n093vg3lhrl";
+  };
+
+  propagatedBuildInputs = [ serpent ];
+
+  disabled = pythonOlder "3.6";
```
Comment (jonringer) [inline pkgs/development/python-modules/pyro5/default.nix]: usually disabled follows version in ordering of attrs

File: pkgs/applications/networking/maestral/default.nix [hunk 6]
```diff
@@ -1,24 +1,38 @@
-{ stdenv, lib, python3Packages, fetchFromGitHub
-, withGui ? false, wrapQtAppsHook ? null }:
+{ stdenv
+, lib
+, python3Packages
```
Comment (jonringer) [inline pkgs/applications/networking/maestral/default.nix]: in general, you should import the interpreter itself, and access the packages through python.pkgs

File: pkgs/development/python-modules/pyro5/default.nix [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+  checkPhase = ''
+    pytest -k '${lib.concatMapStringsSep " and " (e: "not ${e}") ignoredTests}'
+  '';
+
+  meta = with lib; {
+    description = "Distributed object middleware for Python (RPC)";
+    homepage = "https://github.com/irmen/Pyro5";
+    license = licenses.mit;
+    maintainers = with maintainers; [];
```
Comment (jonringer) [inline pkgs/development/python-modules/pyro5/default.nix]: please add yourself as maintainer if you're adding it :)

File: pkgs/applications/networking/maestral/default.nix [hunk 8]
```diff
@@ -1,24 +1,38 @@
...
+    keyrings-alt
+    Pyro5
+    requests
+    u-msgpack-python
+    watchdog
   ] ++ lib.optionals stdenv.isLinux [
-    sdnotify systemd
-  ] ++ lib.optional withGui pyqt5);
+    sdnotify
+    systemd
```
Comment (jonringer) [inline pkgs/applications/networking/maestral/default.nix]: not sure if propagating theses is the right thing to do, personally I would wrap them

**Comment by peterhoeg** [comment]
Thanks for your review @jonringer

File: pkgs/applications/networking/maestral/default.nix (hunk 8 above)
Comment (peterhoeg) [reply pkgs/applications/networking/maestral/default.nix]: Why?

**Pull request merged**